inloop/solutions/management/commands/generate_submissions.py

```diff
@@ -0,0 +1,117 @@
+import random
+from datetime import datetime, timedelta
+from pathlib import Path
+
+from django.contrib.auth import get_user_model
+from django.core.management.base import BaseCommand
+from django.db.transaction import atomic
+from django.utils.timezone import make_aware
+
+from inloop.solutions.models import Solution
+from inloop.tasks.models import Task
+
+User = get_user_model()
+
+file_dir = Path(__file__).resolve().parent
+
+
+class Command(BaseCommand):
+    help = "Generate submissions for demo purposes."
+    dateformat = "%d.%m.%Y"
+
+    def create_users(self, amount):
+        with file_dir.joinpath("animals.txt").open(mode="r") as f:
+            usernames = [animal.strip().lower() for animal in f.readlines()]
+        chosen_names = random.sample(usernames, amount)
+        users = []
+        for name in chosen_names:
+            username = "Anonymous {}".format(name)
+            email = "{}@example.com".format(name)
+            password = "secret"
+            # Since username is a unique constraint, it needs to be
+            # checked first.
+            if User.objects.filter(username=username).exists():
+                users.append(User.objects.get(username=username))
+            else:
+                users.append(User.objects.create_user(
+                    username=username, email=email, password=password
+                ))
+        return users
+
+    def create_solution(self, user, task, amount, start_date, end_date):
+        # Randomly choose, if the solution succeeded
+        passed = random.choice([True, False])
+        # Generate random submission date between start and end date
+        submission_date = start_date + timedelta(
+            seconds=random.randint(0, int((end_date - start_date).total_seconds()))
+        )
+        solution = Solution.objects.create(
+            author=user, task=task, passed=passed,
+        )
+        # If auto_now_add is active, the submission date must be supplied
```

Comment:
What you meant is:
"If auto_now_add is active, the submission date must be supplied *after* object creation." (with stress on after)

What I "read" initially:
"auto_now_add doesn't work"

Proposal:
"Because of auto_now_add, we can't set a custom submission_date through the constructor."

(Reading through https://docs.djangoproject.com/en/2.2/ref/models/fields/#django.db.models.DateField, in retroperspective it probably would have been better if I had used `default=timezone.now()` for this field.)

Reply:
> Proposal:
> "Because of auto_now_add, we can't set a custom submission_date through the constructor."

Your proposal is better and does not imply erroneous behaviour of the `auto_now_add` option.

> (Reading through https://docs.djangoproject.com/en/2.2/ref/models/fields/#django.db.models.DateField, in retroperspective it probably would have been better if I had used `default=timezone.now()` for this field.)

Yes, I can't remember where, but a few weeks ago, I ran into other problems using the `auto_now_add` option, because it automatically inherits `editable=False`.